nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/test/java/org/apache/nifi/json/TestInferJsonSchemaAccessStrategy.java

```diff
@@ -189,14 +191,65 @@ public void testDocsExample() throws IOException {
         assertSame(RecordFieldType.STRING, schema.getDataType("nullValue").get().getFieldType());
     }
 
-    private RecordSchema inferSchema(final File file) throws IOException {
+    @ParameterizedTest(name = "{index} {2}")
+    @MethodSource("skipToArgumentProvider")
+    void testInferenceSkipsToArray(final String jsonPath, final String skipToField, String testName) throws IOException {
+        final File file = new File(jsonPath);
+        final RecordSchema schema = inferSchema(file, skipToField);
+
+        final RecordField field1 = schema.getField("id").get();
+        assertSame(RecordFieldType.INT, field1.getDataType().getFieldType());
+
+        final RecordField field2 = schema.getField("balance").get();
+        assertSame(RecordFieldType.DOUBLE, field2.getDataType().getFieldType());
+    }
+
+    @Test
+    void testInferenceSkipsToSimpleFieldAndNoNestedObjectOrArrayFound() throws IOException {
+        final File file = new File("src/test/resources/json/single-element-nested-array-middle.json");
+        final RecordSchema schema = inferSchema(file, "name");
+
+        assertEquals(0, schema.getFieldCount());
+    }
+
+    @Test
+    void testInferenceSkipsToNonExistentField() {
+        final File file = new File("src/test/resources/json/single-element-nested-array.json");
+        final Exception exception = assertThrows(IOException.class,
+                () -> inferSchema(file, "notfound")
+        );
+        assertEquals("The defined skipTo json field is not found when inferring json schema.", exception.getMessage());
```

Comment:
It is best to avoid assert exact exception messages since it makes tests brittle, and the message should not necessarily be part of the contract.